### PR TITLE
Restructure example notebooks and their thumbnail galleries in docs

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
# Description

This PR:

1. Adds a thumbnail gallery for all sets of notebooks on the main examples page, for easy access
2. Includes the Getting Started and Creating Models sets of notebooks in the user guide

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all`
- [x] The documentation builds: `$ python run-tests.py --doctest`

You can run unit and doctests together at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by agriyakhetarpal** [comment]
Can be viewed at https://pybamm--3078.org.readthedocs.build/en/3078/source/examples/notebooks/index.html

File: docs/source/user_guide/index.md [hunk 1]
```diff
@@ -22,3 +22,15 @@ maxdepth: 2
...
 fundamentals/index
 ```
+
+```{toctree}
+---
+caption: Example notebooks
+maxdepth: 1
+---
+../examples/notebooks/getting_started/index
+../examples/notebooks/creating_models/index
```
Comment (Saransh-cpp) [inline docs/source/user_guide/index.md:32]: Should we also display thumbnails directly in the user guide page?

**Review by Saransh-cpp — APPROVED** [review]
Thanks, @agriyakhetarpal! Looks great!

File: docs/source/user_guide/index.md (hunk 1 above)
Comment (agriyakhetarpal) [reply docs/source/user_guide/index.md]: Good idea, done

**Comment by Saransh-cpp** [comment]
Did this work? I cannot see the changes in the latest docs for some reason.

**Comment by Saransh-cpp** [comment]
Ah, I was expecting the thumbnails to show up at https://pybamm--3078.org.readthedocs.build/en/3078/source/examples/index.html

@agriyakhetarpal is it possible to show thumbnails on this page?

**Comment by agriyakhetarpal** [comment]
That page is supposed to contain two toctree entries pointing to the example notebooks and the scripts, but it only contains the notebooks for now. We can restructure the page again later to contain thumbnails for both when the example scripts are added

**Comment by Saransh-cpp** [comment]
I see, we can discuss the scrips in this week's meet.

**Pull request merged**